### PR TITLE
Added anchors to ega_accession and dbgap_accession patterns. Fixes #1425

### LINKED ISSUE #1425: `ega_accessions` and `dbgap_accessions` patterns are missing anchors
Patterns for `project.ega_accessions` and `project.dbgap_accessions` are missing  line anchors (^$)
https://schema.humancellatlas.org/type/project/15.0.0/project

**To reproduce**

 

Observed inconsistency between accession values in the project index within Azul. Saw the missing anchors within the field pattern.

**Schema versions**

 
project.json schema v15.0.0

**Observed behavior**

Pattern matches the whole string `"EGAS00001003779; EGAS00001003769"`

**Expected behavior**

I expect the pattern to do a full match including line anchors.
`"EGAS00001003779", "EGAS00001003769"`

**Screenshots or Messages**

### PR DESCRIPTION
**Pull request by willrockout**
### Release notes

For `project.json` schema:
- Added anchors to `ega_accessions` and `dbgap_accessions`
 

### Reviews requested

- Need 2 Reviewers to approve because this is a `minor` update
- Last Reviewer to review/approve, merge changes by <deadline>

### CODE REVIEW

**Review by theathorn — APPROVED** [review]

**Review by NoopDog — APPROVED** [review]

**Review by ruchim — COMMENTED** [review]
Quick question — if one HCA project is associated with 2 Dbgap or 2 EGA studies —but they only have only been approved to access one of the multiple studies — that’s complexity for the custodian to manage and to decide whether someone should get access?

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

File: json_schema/update_log.csv [hunk 1]
```diff
@@ -1 +1,2 @@
 Schema,Change type,Change message,Version,Date
+type/project/project.json,patch,"Added anchors to ega_accessions and dbgap_accessions. Fixes #1425",,
```
Comment (ESapenaVentura) [inline json_schema/update_log.csv:2]:
```suggestion
type/project/project,major,"Added anchors to ega_accessions and dbgap_accessions. Fixes #1425",,
```

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
I put a suggestion on the change type. I think we need to review the docs on the patch changes.

This is not a patch change by any mean, as it breaks backwards compatibility with previous instances

(e.g. this is addressing an issue with a field that was filled with 2 ega accessions in a string, that won't be accepted as valid with the new version thus breaking compatibility) https://github.com/HumanCellAtlas/metadata-schema/blob/master/docs/evolution.md#schema-versioning

**Review by ruchim — APPROVED** [review]
looks good to me! lets just talk through implications of this for managed access datasets in a future session

**Review by ESapenaVentura — APPROVED** [review]

**Pull request merged**